apm-sniffer/apm-sdk-plugin/tomcat-7.x-8.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat78x/define/TomcatInstrumentation.java

```diff
@@ -93,4 +93,13 @@ public boolean isOverrideArgs() {
             }
         };
     }
+    @Override
+    protected String[] witnessClasses() {
+        /**
+         * @see <code>javax.servlet.annotation.HandlesTypes<code/>
+         */
+        return new String[] {
+            "javax.servlet.annotation.HandlesTypes"
```

Comment:
Can you sure that tomcat 6 can't run with servlet 3.x ? because the `javax.servlet.annotation.HandlesTypes` class is in `javax.servlet-3.0.jar`. I am not sure that it is works

Reply:
Tomcat 6 work in servlet-api  2.5.

Reply:
I support you need find the witness class in tomcat core jar